### ISSUE TITLE
[tensorflow-cc:x64-windows] build failure

### ISSUE THREAD

**Issue by alexreinking** [microsoft/vcpkg#17578]
**Host Environment**
- OS: Windows x64
- Compiler: MSVC 19.28.29913
- Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

**To Reproduce**
`./vcpkg install tensorflow-cc:x64-windows`

**Failure logs** 
```
D:\vcpkg>vcpkg install tensorflow-cc:x64-windows
Computing installation plan...
The following packages will be built and installed:
    tensorflow-cc[core]:x64-windows -> 2.4.1
  * tensorflow-common[core]:x64-windows -> 2.4.1#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\Alex Reinking\AppData\Local\vcpkg\archives\25\25ba2e73e538d1e14ecc334e9c5ce57ab80ff5f5.zip
Could not locate cached archive: C:\Users\Alex Reinking\AppData\Local\vcpkg\archives\71\713c99c62fcb7821e52e80038b3b5f1b27e4ae65.zip
Starting package 1/2: tensorflow-common:x64-windows
Building package tensorflow-common[core]:x64-windows...
-- Installing: D:/vcpkg/packages/tensorflow-common_x64-windows/share/tensorflow-common/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\Alex Reinking\AppData\Local\vcpkg\archives\25\25ba2e73e538d1e14ecc334e9c5ce57ab80ff5f5.zip
Building package tensorflow-common[core]:x64-windows... done
Installing package tensorflow-common[core]:x64-windows...
Installing package tensorflow-common[core]:x64-windows... done
Elapsed time for package tensorflow-common:x64-windows: 121.6 ms
Starting package 2/2: tensorflow-cc:x64-windows
Building package tensorflow-cc[core]:x64-windows...
-- Downloading https://github.com/bazelbuild/bazel/releases/download/3.7.0/bazel-3.7.0-windows-x86_64.zip -> bazel-3.7.0-windows-x86_64.zip...
-- Downloading https://repo.msys2.org/msys/x86_64/unzip-6.0-2-x86_64.pkg.tar.xz -> msys-unzip-6.0-2-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/patch-2.7.6-1-x86_64.pkg.tar.xz -> msys-patch-2.7.6-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/gzip-1.10-1-x86_64.pkg.tar.xz -> msys-gzip-1.10-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-numpy-1.19.0-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-python-numpy-1.19.0-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openblas-0.3.10-2-any.pkg.tar.zst -> msys-mingw-w64-x86_64-openblas-0.3.10-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc-libgfortran-10.2.0-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-gcc-libgfortran-10.2.0-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-3.8.5-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-python-3.8.5-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-bzip2-1.0.8-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-bzip2-1.0.8-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-bzip2-1.0.8-1-any.pkg.tar.xz... Failed. Status: 22;"HTTP response code said error"
-- Downloading https://www2.futureware.at/~nickoe/msys2-mirror/mingw/x86_64/mingw-w64-x86_64-bzip2-1.0.8-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-bzip2-1.0.8-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpdecimal-2.5.0-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-mpdecimal-2.5.0-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ncurses-6.2-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-ncurses-6.2-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libsystre-1.0.1-4-any.pkg.tar.xz -> msys-mingw-w64-x86_64-libsystre-1.0.1-4-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libtre-git-r128.6fb7206-2-any.pkg.tar.xz -> msys-mingw-w64-x86_64-libtre-git-r128.6fb7206-2-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.g-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-openssl-1.1.1.g-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ca-certificates-20200601-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-ca-certificates-20200601-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-p11-kit-0.23.20-2-any.pkg.tar.xz -> msys-mingw-w64-x86_64-p11-kit-0.23.20-2-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libtasn1-4.16.0-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-libtasn1-4.16.0-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-sqlite3-3.33.0-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-sqlite3-3.33.0-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-readline-8.0.004-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-readline-8.0.004-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-termcap-1.3.1-6-any.pkg.tar.zst -> msys-mingw-w64-x86_64-termcap-1.3.1-6-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-tk-8.6.10-2-any.pkg.tar.zst -> msys-mingw-w64-x86_64-tk-8.6.10-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-tcl-8.6.10-1-any.pkg.tar.xz -> msys-mingw-w64-x86_64-tcl-8.6.10-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-xz-5.2.5-2-any.pkg.tar.zst -> msys-mingw-w64-x86_64-xz-5.2.5-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gettext-0.19.8.1-10-any.pkg.tar.zst -> msys-mingw-w64-x86_64-gettext-0.19.8.1-10-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc-libs-10.2.0-9-any.pkg.tar.zst -> msys-mingw-w64-x86_64-gcc-libs-10.2.0-9-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpc-1.2.0-2-any.pkg.tar.zst -> msys-mingw-w64-x86_64-mpc-1.2.0-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpfr-4.1.0-3-any.pkg.tar.zst -> msys-mingw-w64-x86_64-mpfr-4.1.0-3-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gmp-6.2.0-3-any.pkg.tar.zst -> msys-mingw-w64-x86_64-gmp-6.2.0-3-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-expat-2.2.10-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-expat-2.2.10-1-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libffi-3.3-3-any.pkg.tar.zst -> msys-mingw-w64-x86_64-libffi-3.3-3-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libiconv-1.16-2-any.pkg.tar.zst -> msys-mingw-w64-x86_64-libiconv-1.16-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-zlib-1.2.11-9-any.pkg.tar.zst -> msys-mingw-w64-x86_64-zlib-1.2.11-9-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst -> msys-mingw-w64-x86_64-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst...
-- Using msys root at D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a
-- Downloading https://github.com/tensorflow/tensorflow/archive/v2.4.1.tar.gz -> tensorflow-tensorflow-v2.4.1.tar.gz...
-- Extracting source D:/vcpkg/downloads/tensorflow-tensorflow-v2.4.1.tar.gz
-- Applying patch D:/vcpkg/installed/x64-windows/share/tensorflow-common/fix-build-error.patch
-- Applying patch D:/vcpkg/installed/x64-windows/share/tensorflow-common/fix-windows-build.patch
-- Using source at D:/vcpkg/buildtrees/tensorflow-cc/src/v2.4.1-81b5bb6ede.clean
-- Configuring TensorFlow (dbg)
-- Warning: Building TensorFlow can take an hour or more.
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/usr/bin/bash.exe --noprofile --norc -c "'D:/vcpkg/downloads/tools/bazel/3.7.0-windows/bazel.exe' build --verbose_failures --compilation_mode=dbg --features=fastbuild    --python_path='D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/mingw64/bin/python3.exe' --define=no_tensorflow_py_deps=true ///tensorflow:tensorflow_cc.dll ///tensorflow:install_headers"
    Working Directory: D:/vcpkg/buildtrees/tensorflow-cc/x64-windows-dbg
    See logs for more information:
      D:\vcpkg\buildtrees\tensorflow-cc\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  installed/x64-windows/share/tensorflow-common/tensorflow-common.cmake:222 (vcpkg_execute_build_process)
  ports/tensorflow-cc/portfile.cmake:7 (include)
  scripts/ports.cmake:142 (include)


Error: Building package tensorflow-cc:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: tensorflow-cc:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
```

[build-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/6400985/build-x64-windows-dbg-err.log)

**Comment by JackBoosY** [comment]
```
[0 / 1,469] [Prepa] BazelWorkspaceStatusAction stable-status.txt
ERROR: D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_alex reinking/hwfhzki5/external/llvm-project/llvm/BUILD:46:18: Executing genrule @llvm-project//llvm:config_gen failed (Exit 2): bash.exe failed: error executing command 
  cd D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_alex reinking/hwfhzki5/execroot/org_tensorflow
  SET PATH=D:\vcpkg\downloads\tools\msys2\b98a6bfb7dc2001a\usr\bin;D:\vcpkg\downloads\tools\msys2\b98a6bfb7dc2001a\bin;C:\WINDOWS;C:\WINDOWS\System32;C:\WINDOWS\System32\WindowsPowerShell\v1.0
    SET PYTHON_BIN_PATH=D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/mingw64/bin/python3.exe
    SET PYTHON_LIB_PATH=D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/mingw64/lib/python3.8/site-packages
    SET RUNFILES_MANIFEST_ONLY=1
    SET TF2_BEHAVIOR=1
    SET TF_CONFIGURE_IOS=0
  D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/usr/bin/bash.exe -c source external/bazel_tools/tools/genrule/genrule-setup.sh; bazel-out/x64_windows-dbg/bin/third_party/llvm/expand_cmake_vars.exe "ENABLE_BACKTRACES=1" "LLVM_BINDIR=/dev/null" "LLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING=0" "LLVM_ENABLE_ABI_BREAKING_CHECKS=0" "LLVM_ENABLE_THREADS=0" "LLVM_ENABLE_ZLIB=1" "LLVM_HAS_ATOMICS=1" "LLVM_INCLUDEDIR=/dev/null" "LLVM_INFODIR=/dev/null" "LLVM_MANDIR=/dev/null" "LLVM_NATIVE_TARGET=1" "LLVM_NATIVE_TARGETINFO=1" "LLVM_NATIVE_TARGETMC=1" "LLVM_NATIVE_ASMPRINTER=1" "LLVM_NATIVE_ASMPARSER=1" "LLVM_NATIVE_DISASSEMBLER=1" "LLVM_PREFIX=/dev/null" "LLVM_VERSION_MAJOR=0" "LLVM_VERSION_MINOR=0" "LLVM_VERSION_PATCH=0" "PACKAGE_NAME=llvm" "PACKAGE_STRING=llvm tensorflow-trunk" "PACKAGE_VERSION=tensorflow-trunk" "RETSIGTYPE=void" "LLVM_HOST_TRIPLE=x86_64-pc-win32" "LLVM_DEFAULT_TARGET_TRIPLE=x86_64-pc-win32" "LLVM_NATIVE_ARCH=X86" "HAVE_ERRNO_H=1" "HAVE_EXECINFO_H=1" "HAVE_FCNTL_H=1" "HAVE_FENV_H=1" "HAVE_INTTYPES_H=1" "HAVE_MALLOC_H=1" "HAVE_SIGNAL_H=1" "HAVE_STDINT_H=1" "HAVE_SYS_STAT_H=1" "HAVE_SYS_TYPES_H=1" "HAVE_ZLIB_H=1" "BACKTRACE_HEADER=execinfo.h" "HAVE_GETCWD=1" "HAVE_INT64_T=1" "HAVE_STRERROR=1" "HAVE_STRTOLL=1" "HAVE_SYSCONF=1" "HAVE_UINT64_T=1" "HAVE__CHSIZE_S=1" "HAVE___CHKSTK=1" "stricmp=_stricmp" "strdup=_strdup" "LTDL_SHLIB_EXT=.dll"< external/llvm-project/llvm/include/llvm/Config/config.h.cmake > bazel-out/x64_windows-dbg/bin/external/llvm-project/llvm/include/llvm/Config/config.h
Execution platform: @local_execution_config_platform//:platform
D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/mingw64/bin/python3.exe: can't open file 'D:\vcpkg\buildtrees\tensorflow-cc\.bzl\_bazel_alex': [Errno 2] No such file or directory
```
Please remove folder _D:/vcpkg/buildtrees/tensorflow-cc_ and try again.

Thanks.

**Comment by mxy493** [comment]
Logs: 

```
D:\vcpkg  master ≡  1m 17.383s
❯ vcpkg install tensorflow-cc:x64-windows
Computing installation plan...
The following packages will be built and installed:
    tensorflow-cc[core]:x64-windows -> 2.4.1
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\mxy\AppData\Local\vcpkg\archives\62\62ad7fbddbcf229fe98906c6b3f2353e416958ba.zip
Starting package 1/1: tensorflow-cc:x64-windows
Building package tensorflow-cc[core]:x64-windows...
-- Using msys root at D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a
-- Using cached D:/vcpkg/downloads/tensorflow-tensorflow-v2.4.1.tar.gz
-- Extracting source D:/vcpkg/downloads/tensorflow-tensorflow-v2.4.1.tar.gz
-- Applying patch D:/vcpkg/installed/x64-windows/share/tensorflow-common/fix-build-error.patch
-- Applying patch D:/vcpkg/installed/x64-windows/share/tensorflow-common/fix-windows-build.patch
-- Using source at D:/vcpkg/buildtrees/tensorflow-cc/src/v2.4.1-81b5bb6ede.clean
-- Configuring TensorFlow (dbg)
-- You are using HTTPS_PROXY. In case you encounter bazel certificate errors, you might want to set: BAZEL_CUSTOM_CACERTS=/path/to/trust.store (and optionally BAZEL_CUSTOM_CACERTS_PASSWORD), and to enable vcpkg to actually use it: VCPKG_KEEP_ENV_VARS=BAZEL_CUSTOM_CACERTS;BAZEL_CUSTOM_CACERTS_PASSWORD
-- (For BAZEL_CUSTOM_CACERTS please use forward slashes instead of backslashes on Windows systems.
-- Warning: Building TensorFlow can take an hour or more.
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/usr/bin/bash.exe --noprofile --norc -c "'D:/vcpkg/downloads/tools/bazel/3.7.0-windows/bazel.exe' build --verbose_failures --compilation_mode=dbg --features=fastbuild    --python_path='D:/vcpkg/downloads/tools/msys2/b98a6bfb7dc2001a/mingw64/bin/python3.exe' --define=no_tensorflow_py_deps=true ///tensorflow:tensorflow_cc.dll ///tensorflow:install_headers"
    Working Directory: D:/vcpkg/buildtrees/tensorflow-cc/x64-windows-dbg
    See logs for more information:
      D:\vcpkg\buildtrees\tensorflow-cc\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  installed/x64-windows/share/tensorflow-common/tensorflow-common.cmake:222 (vcpkg_execute_build_process)
  ports/tensorflow-cc/portfile.cmake:7 (include)
  scripts/ports.cmake:141 (include)


Error: Building package tensorflow-cc:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: tensorflow-cc:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
```

And: [build-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/6525910/build-x64-windows-dbg-err.log)

How can I solve this problem, please?

**Comment by JackBoosY** [comment]
```
ERROR: D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_mxy/hwfhzki5/external/local_execution_config_python/BUILD:69:8: @local_execution_config_python//:python_include: illegal output file name 'python_includeActive code page: 65001' in rule @local_execution_config_python//:python_include: invalid target name 'python_includeActive code page: 65001': target names may not contain ':'
ERROR: D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_mxy/hwfhzki5/external/local_execution_config_python/BUILD:69:8: @local_execution_config_python//:python_include: missing value for mandatory attribute 'outs' in 'genrule' rule
ERROR: D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_mxy/hwfhzki5/external/local_execution_config_python/BUILD:243:8: @local_execution_config_python//:numpy_include: illegal output file name 'numpy_include/numpyActive code page: 65001' in rule @local_execution_config_python//:numpy_include: invalid target name 'numpy_include/numpyActive code page: 65001': target names may not contain ':'
ERROR: D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_mxy/hwfhzki5/external/local_execution_config_python/BUILD:243:8: @local_execution_config_python//:numpy_include: missing value for mandatory attribute 'outs' in 'genrule' rule
ERROR: Analysis of target '//tensorflow:tensorflow_cc.dll' failed; build aborted: error loading package '@local_execution_config_python//': Package '' contains errors
```

Should be related to the system charset, please enable utf-8 support in Windows.
see https://www.quora.com/Why-doesnt-Microsoft-use-UTF-8-on-Windows-10.

**Comment by jgehw** [comment]
@mxy493 The error you're facing is something totally different (though the error message in the terminal looked quite similar). In case the utf-8 setting suggested by @JackBoosY didn't already solve your error, can you please check the file `D:/vcpkg/buildtrees/tensorflow-cc/.bzl/_bazel_mxy/hwfhzki5/external/local_execution_config_python/BUILD`? Does it contain any invalid paths? In particular is there `Active code page: 65001` appended to any path name, especially around lines 69ff or 243ff? And what are your Windows language settings?

**Comment by jgehw** [comment]
@mxy493 PS: the output `Active code page: 65001` seems to be generated by the Windows command `chcp`. What does `chcp` show when run in the terminal before calling `vcpkg`? In case it shows a different code page then 65001, could you try if setting `chcp 65001` before calling `vcpkg` solves the issue? (Just guessing: it could be that the default code page is something unsupported and when `bazel` attempts to fix it by changing the code page it produces this additional output which is not correctly discarded but redirected into some variable / config file.)

**Comment by mxy493** [comment]
@jgehw I think my system's charset is GBK, and I have changed the charset of the terminal to UTF-8 by the registry.
And my user name is "mxy", there is no space in the middle.
```
D:\mxy\Desktop
❯ chcp
Active code page: 65001
```
@JackBoosY I didn't try the solution of change the windows charset, because it will cause many applications in my system garbled.

**Comment by mxy493** [comment]
@jgehw In fact, I don’t need to install tensorflow now. But in order to find out the cause, I tried to reproduce this problem today. But it takes up more than 60G of space, my computer does not have enough space😅So it build failed and failed to reproduce the previous problem.

**Comment by jgehw** [comment]
> @jgehw I think my system's charset is GBK, and I have changed the charset of the terminal to UTF-8 by the registry.

@mxy493 That's quite important information - can you please share, what exactly you did to set UTF-8 via registry?

(Just guessing: I suppose because your system default is GBK, Windows automatically applies your registry changes to every terminal session you start (so code page 65001, i.e. UTF-8, is already active when you query it via `chcp`). Apparently the `bazel` build tool at some point causes your registry changes to be re-applied, resulting in the "Active code page: 65001" output, which is redirected into a path variable and then causes the error.)

**Comment by JackBoosY** [comment]
Related: https://github.com/microsoft/vcpkg/issues/12865.

**Comment by mxy493** [comment]
@jgehw I built successfully last night, and I think I found the reason.
I mistakenly wrote `chcp 65001>nul` as `chcp 65001>null` when modifying the registry.

<img width="608" alt="Snipaste_2021-05-27_11-59-06" src="https://user-images.githubusercontent.com/46776216/119763942-1428bd80-bee3-11eb-80c2-94ee2e7f0b59.png">

So it always write `Active code page: 65001` to the file named `null` every time.

**Comment by jgehw** [comment]
I think this issue can be closed now. First error is fixed via PR #18111 and second error needn't be address via vcpkg (was autorun misconfiguration on user machine).

**Comment by JackBoosY** [comment]
We haven't been able to repro this; if more information comes up, or this issue appears again, please reopen.
